From ee647a99f8b6a7aad1e49520127ded97639676ea Mon Sep 17 00:00:00 2001
From: Kaarle Ritvanen <kaarle.ritvanen@datakunkku.fi>
Date: Tue, 10 Dec 2024 12:39:07 +0200
Subject: [PATCH] setup: make npm build optional

The distribution package does not contain package.json nor gulpfile.js
and ideally would not depend on access to the npm registry.
---
 setup.py | 9 +++++++--
 1 file changed, 7 insertions(+), 2 deletions(-)

diff --git a/setup.py b/setup.py
index 89a21019a..dfe95ba73 100755
--- a/setup.py
+++ b/setup.py
@@ -22,8 +22,13 @@ from oscar import get_version  # noqa isort:skip
 
 class BuildNPM(build_module.build):
     def run(self):
-        subprocess.check_call(["npm", "install"])
-        subprocess.check_call(["npm", "run", "build"])
+        try:
+            os.stat(os.path.join(PROJECT_DIR, "package.json"))
+        except FileNotFoundError:
+            pass
+        else:
+            subprocess.check_call(["npm", "install"])
+            subprocess.check_call(["npm", "run", "build"])
         super().run()
 
 
-- 
2.47.1

